packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -195,12 +215,70 @@ export class CronjobController extends BaseController<
       `${controllerName}:getBackgroundEvents`,
       (...args) => this.getBackgroundEvents(...args),
     );
+  }
 
-    this.dailyCheckIn().catch((error) => {
-      logError(error);
+  /**
+   * Execute a request.
+   *
+   * @param snapId - ID of a Snap.
+   * @param request - Request to be executed.
+   */
+  async #executeRequest(snapId: SnapId, request: unknown) {
+    await this.messagingSystem
+      .call('SnapController:handleRequest', {
+        snapId,
+        origin: '',
+        handler: HandlerType.OnCronjob,
+        request: request as Record<string, unknown>,
+      })
+      .catch((error) => {
+        logError(error);
+      });
+  }
+
+  /**
+   * Setup a timer.
+   *
+   * @param id - ID of a timer.
+   * @param ms - Time in milliseconds.
+   * @param snapId - ID of a Snap.
+   * @param onComplete - Callback function to be executed when the timer completes.
+   * @param validateTime - Whether the time should be validated.
```

Comment:
What's the use case for this not being validated?

Reply:
For scheduling cronjobs, the time validation is done in the `schedule` function.

Reply:
Where exactly?

Reply:
https://github.com/MetaMask/snaps/pull/3029/files#diff-d3d7c575b1031aeae16afee89d5c29a141c7edc55f190c5c0bc8d0071e9fcf2fR345-R353

Reply:
Maybe I don't misunderstand, but this seems to be a different kind of validation. `#schedule` checks if the job should be scheduled, and is not too far into the future, whereas `#setupTimer` checks if the scheduled time is not negative.

Regardless, I feel like we may as well always check if the scheduled time is not negative, since the performance impact is minimal. Unless a negative time means it should be run immediately?

Reply:
We'd never have a negative time since cronjobs can't ever be in the past, the only check it needs to make is the daily timeout check. And yeah this was to avoid making an unnecessary check, but I can remove if you think it's confusing or something?

Reply:
Yeah, I'd just remove it.